components/webln/index.js

```diff
@@ -3,29 +3,39 @@ import { LNbitsProvider, useLNbits } from './lnbits'
 import { NWCProvider, useNWC } from './nwc'
 import { useToast, withToastFlow } from '@/components/toast'
 import { gql, useMutation } from '@apollo/client'
+import { LNCProvider, useLNC } from './lnc'
 
 const WebLNContext = createContext({})
 
 const syncProvider = (array, provider) => {
   const idx = array.findIndex(({ name }) => provider.name === name)
+  const enabled = [Status.Enabled, Status.Locked].includes(provider.status)
   if (idx === -1) {
     // add provider to end if enabled
-    return provider.enabled ? [...array, provider] : array
+    return enabled ? [...array, provider] : array
   }
   return [
     ...array.slice(0, idx),
     // remove provider if not enabled
-    ...provider.enabled ? [provider] : [],
+    ...enabled ? [provider] : [],
     ...array.slice(idx + 1)
   ]
 }
 
 const storageKey = 'webln:providers'
 
+export const Status = {
+  Initialized: 'Initialized',
+  Enabled: 'Enabled',
+  Locked: 'Locked',
+  Error: 'Error'
+}
+
```

Comment:
I replaced the use of a state pair `(initialized, enabled)` with this enum like state for a wallet's status.